### PR TITLE
Fix docker to containerd migration when upgrading to k8s 1.24

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Fix kubelet fails to start with "Error: failed to parse kubelet flag: unknown flag: --network-plugin" when migrating from containerd to docker and upgrading kubernetes to 1.24.x.

Various other improvements around the migration from Docker to Containerd.

#### Which issue(s) this PR fixes:

Fixes NONE

#### Does this PR introduce a user-facing change?

```release-note
- Fixes an issue that prevents upgrading Kubernetes to 1.24.x if the CRI has previously been migrated from Docker to Containerd.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: scripts/upgrade.sh [hunk 1]
```diff
@@ -116,10 +116,6 @@ function main() {
     download_util_binaries
     get_machine_id
     merge_yaml_specs
-    # Parse yaml into bash variables so we can get final kubernetes version for cases where we are upgrading 2 minor versions
```
Comment (emosbaugh) [inline scripts/upgrade.sh:119]: [This code](https://github.com/replicatedhq/kurl/commit/4d41a44e75ddc606eddc10553740dd77832b1060) is no longer necessary as it was fixed by [this commit](https://github.com/replicatedhq/kURL/commit/a6bcc0cbdc762e539134387475c4be330bc38843).

File: scripts/Manifest [hunk 2]
```diff
@@ -8,5 +8,5 @@ REPLICATED_APP_URL=https://replicated.app
 KURL_UTIL_IMAGE=replicated/kurl-util:alpha
 KURL_BIN_UTILS_FILE=
 # STEP_VERSIONS array is generated by the server and injected at runtime based on supported k8s versions
-STEP_VERSIONS=("" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "1.15.3" "1.16.4" "1.17.13" "1.18.20" "1.19.15", "1.20.11")
-INSTALLER_YAML=
\ No newline at end of file
+STEP_VERSIONS=(0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 0.0.0 1.16.4 1.17.13 1.18.20 1.19.16 1.20.15 1.21.14 1.22.13 1.23.10 1.24.4)
```
Comment (laverya) [inline scripts/Manifest:11]: I don't think this should have been updated

**Review by laverya — APPROVED** [review]

File: scripts/Manifest (hunk 2 above)
Comment (emosbaugh) [reply scripts/Manifest:11]: It's unrelated but it is helpful for the next developer working on upgrades of kubernetes.

Comment (rrpolanco) [reply scripts/Manifest:11]: @emosbaugh Why do we need all the `0.0` versions?

Comment (emosbaugh) [reply scripts/Manifest:11]: The relevant code is [here](https://github.com/replicatedhq/kURL/blob/7376cc9d7441f22b650d22209be9eec8ab223011/scripts/common/discover.sh#L247-L249). I needed to add the newer versions. This is placeholder for each minor version. 0.0.0 isn't necessary its just what I copied from production. See [here](view-source:https://k8s.kurl.sh/latest)

Comment (ricardomaraschini) [reply scripts/Manifest:11]: @emosbaugh may be ok but: have you intentionally skipped 1.15.3 ?

**Review by ricardomaraschini — APPROVED** [review]

**Pull request merged**